igor-web/src/main/java/com/netflix/spinnaker/igor/health/PollingMonitorHealth.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License')
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.igor.health;
+
+import com.netflix.spectator.api.Registry;
+import com.netflix.spectator.api.patterns.PolledMeter;
+import com.netflix.spinnaker.igor.polling.PollingMonitor;
+import org.joda.time.DateTimeConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.boot.actuate.health.Status;
+import org.springframework.context.ApplicationContext;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Predicate;
+
+import static java.util.stream.Collectors.toList;
+
+/**
+ * The health status is based on whether the poller is running and how long ago it last polled. If 5 times the polling
+ * interval has passed since the last poll the poller is considered _down_.
+ */
+@Component
+public class PollingMonitorHealth implements HealthIndicator {
+    private static Logger log = LoggerFactory.getLogger(PollingMonitorHealth.class);
+    private final ApplicationContext applicationContext;
+    private AtomicBoolean upOnce;
+    private final AtomicLong downPollers;
+
+    @Autowired
+    public PollingMonitorHealth(Registry registry, ApplicationContext applicationContext) {
+        this.applicationContext = applicationContext;
+        this.upOnce = new AtomicBoolean(false);
+        this.downPollers = PolledMeter.using(registry)
+            .withName("health.pollers.down")
+            .monitorValue(new AtomicLong(0));
+    }
+
+    @Override
+    public Health health() {
+        final Collection<PollingMonitor> pollers = getPollingMonitors();
+        List<PollingMonitor> downPollingMonitors = pollers
+            .stream()
+            .filter(hasBeenFiveTimesPollIntervalSinceLastPollPredicate())
+            .collect(toList());
+
+        if (upOnce.get() && !downPollingMonitors.isEmpty()) {
```

Comment:
This seems like it is going to initially fall through to returning UP before upOnce has been set to true

Not sure we want that since I think we want to make sure we have had one successful poll of each before we initially go into service

Reply:
Chatted with @cfieber offline about the circular dependency between poller's health and discovery status. This fixes that.
The plan is to add some additional health checks per poller and that will come in another PR.